src/samples/audio/smart_amp_test_llext/CMakeLists.txt

```diff
@@ -80,8 +80,8 @@ add_custom_command(OUTPUT lib${MODULE}_out.so
 	DEPENDS ${MODULE}
 	COMMAND ${SOF_BASE}scripts/llext_link_helper.py
 		-f lib${MODULE}.so -t "0xa06ca000" ${CMAKE_C_COMPILER} --
-		${MODULE_LINKER_PARAMS} ${EXTRA_LINKED_PARAMS} -fPIC
-		-o ${MODULE}_llext.so $<TARGET_OBJECTS:${MODULE}>
+		${MODULE_LINKER_PARAMS} ${EXTRA_LINKER_PARAMS} -fPIC
+		-o ${MODULE}_pre.so $<TARGET_OBJECTS:${MODULE}>
```

Comment:
problem statement/commit message:

a) why do we need to sign modules?
b) what are the rules for selecting the key used by different modules?

Reply:
@plbossart here the same key is used as the one for the base firmware. In principle any key, accepted by the DSP can be used. Why - for security, for the same reason why we sign the base firmware

Reply:
@lyakh we do need the ability to specify that alternate key for modules that are not part of the base firmware.

Actually maybe it's more complicated even, e.g. if Intel releases modules then presumably they are signed with the same key than the base firmware, but a 3rd party would be required to use a separate key, so the key would need to be specified on a module-by-module basis.

It's not 9am and that already gives me a migraine.

Reply:
> > so the key would need to be specified on a module-by-module basis.
> 
> Right.

@plbossart @marc-hb would 3rd parties even use the xtensa-build-zephyr.sh script to build their modules?

Reply:
> @plbossart @marc-hb would 3rd parties even use the xtensa-build-zephyr.sh script to build their modules?

_You_ tell us! Either in some documentation or in the source code. Then everyone can review and agree or disagree. At the end we're all on the same page.

Reply:
@marc-hb I feel reaching a bit beyond the scope of this work, but let's say - some will and some won't. Those who won't - they'll work out their own way of doing this.
Those who will, they will integrate their modules into their local copy of sof.git in order to be able to use the script. They might or might not want to build both a base image *for the distribution* and any modules within the same build. If they do - I see no reason for them to use different keys. If they don't, i.e. if they only want to build their modules to be passed to an integrator, who will build a base image with their own key - in that case they anyway will execute a complete build, but will only use their modules from it and discard the rest. Either way I so far see no case when it would be essential for anyone to be able to run a single build while signing different parts of it with different keys. If for some unimaginable reason somebody has to do that, I'd say, that this is unsupported which unfortunately for them will mean, that they have to build multiple times - once with each key. And I really don't see this as an insurmountable problem.

Reply:
ack - beyond scope of this PR. All will be signed with a single key initially.

Reply:
> in that case they anyway will execute a complete build, but will only use their modules from it and discard the rest. Either way I so far see no case when it would be essential for anyone to be able to run a single build while signing different parts of it with different keys. 

Great, please just explain this in sof-docs and/or the comments (whichever is more appropriate wherever) and then we can start actually reviewing the implementation.


> And I really don't see this as an insurmountable problem.

I don't think anyone sees any "insurmoutable problem" yet for the simple reason that no one besides (maybe) you seems to have any idea yet how this is going to be used. Which is quite extraordinary when you consider all the time, code and effort you've already put into this.

https://www.atlassian.com/agile/project-management/user-stories